x/inflation/v1/keeper/integration_test.go

```diff
@@ -147,113 +201,144 @@ var _ = Describe("Inflation", Ordered, func() {
 				})
 
 				It("should allocate funds to the community pool", func() {
-					pool, err := s.app.DistrKeeper.FeePool.Get(s.ctx)
+					res, err := s.handler.GetCommunityPool()
 					Expect(err).To(BeNil())
-					balanceCommunityPool := pool.CommunityPool
+					balanceCommunityPoolAmt := res.Pool.AmountOf(denomMint)
 
-					provision := s.app.InflationKeeper.GetEpochMintProvision(s.ctx)
-					params := s.app.InflationKeeper.GetParams(s.ctx)
+					provision := s.network.App.InflationKeeper.GetEpochMintProvision(s.network.GetContext())
+					params := s.network.App.InflationKeeper.GetParams(s.network.GetContext())
 					distribution := params.InflationDistribution.CommunityPool
 					expected := provision.Mul(distribution)
 
-					Expect(balanceCommunityPool.IsZero()).ToNot(BeTrue())
-					Expect(balanceCommunityPool.AmountOf(denomMint).GT(expected)).To(BeTrue())
+					allocatedAmt := balanceCommunityPoolAmt.Sub(prevCommPoolBalanceAmt)
+					Expect(allocatedAmt.IsZero()).ToNot(BeTrue())
+					Expect(allocatedAmt.GT(expected)).To(BeTrue())
 				})
 			})
 		})
 
 		Context("with inflation param enabled", func() {
 			BeforeEach(func() {
-				params := s.app.InflationKeeper.GetParams(s.ctx)
+				params := s.network.App.InflationKeeper.GetParams(s.network.GetContext())
 				params.EnableInflation = true
-				s.app.InflationKeeper.SetParams(s.ctx, params) //nolint:errcheck
+				err := integrationutils.UpdateInflationParams(
+					integrationutils.UpdateParamsInput{
+						Tf:      s.factory,
+						Network: s.network,
+						Pk:      s.keyring.GetPrivKey(0),
+						Params:  params,
+					},
+				)
+				Expect(err).ToNot(HaveOccurred(), "error while setting params")
```

Comment:
we should not rely on using `network.App` in integration tests


```suggestion
				inflationClient := s.network.GetInflationClient()
				paramsRes, err := inflationClient.Params(s.network.GetContext(), &types.QueryParamsRequest{})
				Expect(err).ToNot(HaveOccurred(), "error while getting params")
				updatedParams := paramsRes.Params
				updatedParams.EnableInflation = true
				err = integrationutils.UpdateInflationParams(
					integrationutils.UpdateParamsInput{
						Tf:      s.factory,
						Network: s.network,
						Pk:      s.keyring.GetPrivKey(0),
						Params:  updatedParams,
					},
				)
				Expect(err).ToNot(HaveOccurred(), "error while setting params")
```